README.md

```diff
@@ -12,169 +14,140 @@ For more info and details, please read this [article](https://hackmd.io/@REJeq0M
-      -s, --silent                     Print no informative logs, except compilation results if --log is used
-      -h, --help                       display help for command
+## dotrain_lsp crate
+To get started, install the package:
+```bash
+cargo add dotrain-lsp
+```
 
-<br>
-rainconfig information:
+### Features
+- `js-api`: includes wrappers around main structs and functionalities to provide an API through [wasm-bindgen](https://rustwasm.github.io/docs/wasm-bindgen/) (this feature is always enabled when building for wasm family targets)
 
-    Description:
-    rainconfig.json provides configuration details and information required for .rain compiler.
+```rust
+use std::sync::{Arc, RwLock};
+use dotrain::{Store, RainLanguageServices, TextDocumentItem, LanguageServiceParams, RainDocument};
 
-    usually it should be placed at the root directory of the working workspace and named as 
-    'rainconfig.json' or '.rainconfig.json', as by doing so it will automatically be read 
-    and having rainlang vscode extension, it will provide autocomplete and information on 
-    each field, however if this is not desired at times, it is possible to pass any path for 
-    rainconfig when using the dotrain command using --config option.
+// instantiate arc locked Store
+let meta_store = Arc::new(RwLock::new(Store::default()));
 
-    all fields in the rainconfig are optional and are as follows:
+// the following needs 'lsp' feature to be enabled
+let lang_params = LanguageServiceParams {
+  meta_store: Some(meta_store)
+}
 
-    - src: Specifies list of .rain source files mappings for compilation, where specified 
-    .rain input files will get compiled and results written into output json file.
+// a LSP TextdocumentItem
+let text_document = TextDocumentItem {
+  uri,
+  text,
+  version: 0,
+  language_id: "rainlang".to_string()
+}
 
-    - include: Specifies a list of directories (files/folders) to be included and watched. 
-    'src' files are included by default and folders will be watched recursively for .rain files. 
-    These files will be available as dotrain meta in the cas so if their hash is specified in a
-    compilation target they will get resolved.
+// instantiate RainLanguageServices
+let lang_services = RainLanguageServices::new(lang_params);
 
-    - subgraphs: Additional subgraph endpoint URLs to include when searching for metas of 
-    specified meta hashes in a rainlang document.
+// get LSP diagnostics of the given text document
+let diagnostics = lang_services.do_validate(&text_document, true);
+```
 
-    - meta: Lis of paths (or object of path and hash) of local meta files as binary or utf8 
-    encoded text file containing hex string starting with 0x. Binary meta files should go 
-    under 'meta.binary' field and hex meta files should go under 'meta.hex' field.
+## CLI
+The CLI app can be built form the source (read next section) or be installed with the following command:
```

Comment:
we use nix, lets make the instructions for nix, which should be something like 

`nix run github:rainlanguage/dotrain -- --help`

additionally we should mention that all of this is available as subcommands from the main cli repo, which is also distributed via nix